mujoco_ros2_control/src/mujoco_ros2_control.cpp

```diff
@@ -8,17 +8,23 @@ namespace mujoco_ros2_control
 {
 class MJResourceManager : public hardware_interface::ResourceManager{
 public:
-  MJResourceManager(rclcpp::Node::SharedPtr& node, mjModel* mj_model, mjData* mj_data)
+  MJResourceManager(rclcpp::Node::SharedPtr& node, mjModel*& mj_model, mjData*& mj_data)
```

Comment:
RM will be managed by `ControllerManager` as a unique_ptr, which means we can't get the resource from RM directly without override `ControllerManager`. So I use **`T&*`** to avoid additional operation to get `mj_data` and `mj_model` from RM. And now `MujocoRos2Control` can share the resources with RM.

Reply:
```suggestion
  MJResourceManager(const rclcpp::Node::SharedPtr& node, mjModel*& mj_model, mjData*& mj_data)
```
May i want you to change this? Additionaly if you apply my suggestion, you need to be careful about the order of initialization and base class should be first. I have tried your PR today.